src/TropicalGeometry/matrix.jl

```diff
@@ -29,15 +31,8 @@ julia> det(A)
 (5)
 ```
 """
-function det(A::Generic.MatSpaceElem{R}) where {R<:Union{TropicalSemiringElem,MPolyRingElem{<:TropicalSemiringElem},PolyRingElem{<:TropicalSemiringElem}}}
-    detA = zero(base_ring(A))
-    nrows(A)!=ncols(A) && return detA # return tropical zero if matrix not square
-    for sigma in AbstractAlgebra.SymmetricGroup(nrows(A)) # otherwise follow Leibniz Formula
-        detA += prod([ A[i,sigma[i]] for i in 1:nrows(A) ])
-    end
-    return detA
-end
-
-function det(A::Matrix{R}) where {R<:Union{TropicalSemiringElem,MPolyRingElem{<:TropicalSemiringElem},PolyRingElem{<:TropicalSemiringElem}}}
-    return det(matrix(parent(first(A)),A))
+function det(A::Generic.MatSpaceElem{<:TropicalSemiringElem})
+    nrows(A)!=ncols(A) && error("Non-square matrix")
+    T = base_ring(A)
+    return T(Polymake.tropical.tdet(A))
 end
```

Comment:
```suggestion
end

function det(A::Matrix{TropicalSemiringElem}) 
    return det(matrix(parent(first(A)),A))
end
```

We need a function like this for julia matrices over tropical semirings.  This is causing the error in the tests, I believe.